core/src/main/java/org/jobrunr/scheduling/interval/Interval.java

```diff
@@ -0,0 +1,70 @@
+package org.jobrunr.scheduling.interval;
+
+import org.jobrunr.scheduling.Schedule;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.time.ZoneId;
+
+public class Interval extends Schedule {
+
+    private final Duration duration;
+
+    public Interval(Duration duration) {
+        this.duration = duration;
+    }
+
+    public Interval(String durationExpression) {
+        this.duration = Duration.parse(durationExpression);
+    }
+
+    @Override
+    public Instant next(Instant baseInstant, ZoneId zoneId) {
+        Duration durationUntilNow = Duration.between(baseInstant, Instant.now());
+        long amountOfDurationsUntilNow = durationUntilNow.toNanos() / duration.toNanos();
+
+        return baseInstant.plusNanos(duration.toNanos() * (amountOfDurationsUntilNow + 1));
+    }
```

Comment:
Let's try this for testing:

```java

    @Override
    public Instant next(Instant baseInstant, ZoneId zoneId) {
        return next(baseInstant, Instant.now(), zoneId);
    }

    @VisibleFor("testing")
    Instant next(Instant baseInstant, Instant currentInstant, ZoneId zoneId) {
        Duration durationUntilNow = Duration.between(baseInstant, currentInstant);
        long amountOfDurationsUntilNow = durationUntilNow.toNanos() / duration.toNanos();

        return baseInstant.plusNanos(duration.toNanos() * (amountOfDurationsUntilNow + 1));
    }

```

Now, you can use the `next(Instant baseInstant, Instant currentInstant, ZoneId zoneId)` in the IntervalTest. The reason I'm so picky about this is that in my eyes, otherwise the `@ParameterizedTest @MethodSource("startInstantDurationAndResultInstant")` test does not add much value but I do like it. Now, we can really test this method to the full extend of its responsibility.